internal/server/receiver_handlers.go

```diff
@@ -344,62 +347,39 @@ func (s *ReceiverServer) annotate(ctx context.Context, resource v1beta1.CrossNam
 	if resource.Namespace != "" {
 		namespace = resource.Namespace
 	}
-	resourceName := types.NamespacedName{
+	objectKey := client.ObjectKey{
 		Namespace: namespace,
 		Name:      resource.Name,
 	}
 
-	switch resource.Kind {
-	case sourcev1.BucketKind:
-		var source sourcev1.Bucket
-		if err := s.kubeClient.Get(ctx, resourceName, &source); err != nil {
-			return fmt.Errorf("unable to read %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-		if source.Annotations == nil {
-			source.Annotations = make(map[string]string)
-		}
-		source.Annotations[meta.ReconcileRequestAnnotation] = metav1.Now().String()
-		if err := s.kubeClient.Update(ctx, &source); err != nil {
-			return fmt.Errorf("unable to annotate %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-	case sourcev1.GitRepositoryKind:
-		var source sourcev1.GitRepository
-		if err := s.kubeClient.Get(ctx, resourceName, &source); err != nil {
-			return fmt.Errorf("unable to read %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-		if source.Annotations == nil {
-			source.Annotations = make(map[string]string)
-		}
-		source.Annotations[meta.ReconcileRequestAnnotation] = metav1.Now().String()
-		if err := s.kubeClient.Update(ctx, &source); err != nil {
-			return fmt.Errorf("unable to annotate %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-	case sourcev1.HelmRepositoryKind:
-		var source sourcev1.HelmRepository
-		if err := s.kubeClient.Get(ctx, resourceName, &source); err != nil {
-			return fmt.Errorf("unable to read %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-		if source.Annotations == nil {
-			source.Annotations = make(map[string]string)
-		}
-		source.Annotations[meta.ReconcileRequestAnnotation] = metav1.Now().String()
-		if err := s.kubeClient.Update(ctx, &source); err != nil {
-			return fmt.Errorf("unable to annotate %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-	case imagev1.ImageRepositoryKind:
-		var source imagev1.ImageRepository
-		if err := s.kubeClient.Get(ctx, resourceName, &source); err != nil {
-			return fmt.Errorf("unable to read %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-		if source.Annotations == nil {
-			source.Annotations = make(map[string]string)
-		}
-		source.Annotations[meta.ReconcileRequestAnnotation] = metav1.Now().String()
-		if err := s.kubeClient.Update(ctx, &source); err != nil {
-			return fmt.Errorf("unable to annotate %s '%s' error: %w", resource.Kind, resourceName, err)
-		}
-	default:
-		return fmt.Errorf("kind '%s not suppored", resource.Kind)
+	apiVersionMap := map[string]string{
+		sourcev1.BucketKind:         "source.toolkit.fluxcd.io/v1beta1",
```

Comment:
We need to get rid of the sourcev1 and imagev1 imports, please hardcode here the Kinds for now.

Reply:
We should use the map only if the `APIVersion` is not specified https://github.com/fluxcd/notification-controller/blob/main/api/v1beta1/reference_types.go#L24